In mainline since 2.6.32

From 56bc15204f8389b17ddc98f85c556fc0d9946f59 Mon Sep 17 00:00:00 2001
From: David S. Miller <davem@davemloft.net>
Date: Wed, 30 Sep 2009 16:12:20 -0700
Subject: [PATCH] net: Make setsockopt() optlen be unsigned.

This provides safety against negative optlen at the type
level instead of depending upon (sometimes non-trivial)
checks against this sprinkled all over the the place, in
each and every implementation.

Based upon work done by Arjan van de Ven and feedback
from Linus Torvalds.

Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/isdn/mISDN/socket.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Index: standalone/drivers/isdn/mISDN/socket.c
===================================================================
--- standalone.orig/drivers/isdn/mISDN/socket.c
+++ standalone/drivers/isdn/mISDN/socket.c
@@ -420,7 +420,7 @@ data_sock_ioctl(struct socket *sock, uns
 }
 
 static int data_sock_setsockopt(struct socket *sock, int level, int optname,
-				char __user *optval, unsigned int len)
+				char __user *optval, int len)
 {
 	struct sock *sk = sock->sk;
 	int err = 0, opt = 0;
